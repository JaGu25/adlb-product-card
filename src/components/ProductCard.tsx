import styles from '../styles/styles.module.css';
import useProduct from '../hooks/useProduct';
import React, { createContext, CSSProperties } from 'react';
import { ProductContextProps, Product, onChangeArgs, InitialValues, ProductCardHandles } from '../interfaces/interfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
    product: Product
    children: (args: ProductCardHandles) => JSX.Element
    className?: string
    style?: CSSProperties
    onChange?: (args: onChangeArgs) => void
    value?: number
    initialValues?: InitialValues
}

export const ProductCard = ({ product, children, className, style, onChange, value, initialValues }: Props) => {
    const { counter, increaseBy, maxCount, isMaxCountReached, reset } = useProduct({ product, onChange, value, initialValues });
    return (
        <Provider value={{
            counter,
            increaseBy,
            product,
            maxCount
        }}>
            <div style={style}
                className={`${styles.productCard} ${className}`} >
                {children({
                    count: counter,
                    isMaxCountReached,
                    product,
                    maxCount: initialValues?.maxCount,
                    increaseBy,
                    reset
                })}
            </div >
        </Provider>
    )
}
