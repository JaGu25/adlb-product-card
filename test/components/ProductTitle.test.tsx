import React from 'react'
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product } from './mocks/product';


describe('ProductTitle', () => {
    it('should renders without crashing with custom title', () => {
        const wrapper = renderer.create(<ProductTitle title='Custom Product' className='custom-class' />)
        expect(wrapper).toMatchSnapshot();
    });

    it('should show a component with product title', () => {
        const wrapper = renderer.create(
            <ProductCard product={product}>
                {
                    () => (
                        <ProductTitle></ProductTitle>
                    )
                }
            </ProductCard >
        )
        expect(wrapper).toMatchSnapshot();
    });
});
