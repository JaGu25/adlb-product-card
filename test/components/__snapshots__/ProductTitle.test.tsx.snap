// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should renders without crashing with custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle should show a component with product title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;
